edk2toollib/uefi/authenticated_variables_structure_support.py

```diff
@@ -799,7 +799,7 @@ def PopulateFromFileStream(self, fs):
         Daylight = struct.unpack("<B", fs.read(1))[0]
         fs.seek(1, 1)  # seek past pad2
 
-        self.Time = datetime.datetime(Year, Month, Day, Hour, Minute, Second, NanoSecond / 1000)
+        self.Time = datetime.datetime(Year, Month, Day, Hour, Minute, Second, NanoSecond)
```

Comment:
The behavior of '/' changed between python2 and python3. Python2 would try to infer the data type of the result from the numerator and denominator, whereas in python3 '/' means float division, and '//' means integer division.
 
 In order to correctly pass microseconds to datetime.datetime in python3 the following change would be the easiest way:
 
 
```python3
self.Time = datetime.datetime(Year, Month, Day, Hour, Minute, Second, NanoSecond // 1000)
```